src/components/Card.js

```diff
@@ -9,9 +12,17 @@
       this._cardElement.querySelector('.card__link').addEventListener('click', () => {
         this._handleLikeButton();
       });
+      
       this._cardElement.querySelector('.card__delete-btn').addEventListener('click', () => {
-        this._handleDeleteCard();
+        const deleteCardPopup = createPopupWithYes('#delete-card-modal', () => {
+          api.deleteCard(this._data._id);
+          this._handleDeleteCard();
+        });
+  
+        deleteCardPopup.open();
+        deleteCardPopup.setEventListeners();
```

Comment:
Every popup should be created only one time and `setEventListeners` should be called also once, otherwise you add listeners again and again to the same elements. This can cause a **`memory leak`**